src/app/tests/TestCommandInteraction.cpp

```diff
@@ -1554,6 +1554,43 @@ TEST_F(TestCommandInteraction, TestCommandSenderCommandAsyncSuccessResponseFlow)
     EXPECT_EQ(GetExchangeManager().GetNumActiveExchanges(), 0u);
 }
 
+TEST_F(TestCommandInteraction, CommandSenderDeletedWhenResponseIsPending)
```

Comment:
Self blocking comment to prevent automerge. I want to consult with a few people first